### ISSUE TITLE
MoBi 7.4 secretly changes project file

### ISSUE THREAD

**Issue by UlrichSi** [Open-Systems-Pharmacology/MoBi#356]
If a project created with a previous version of MoBi (7.1 or earlier) is opened in version 7.4, some conversions are automatically performed. MoBi then alters the project file _behind the user's back_, even without explicit **Save** command. This is somewhat annoying when working with a versioning system like subversion. Also, project files might be corrupted in case the computer crashes or other faults during this process.

Generally, I would strongly recommend that user project and data files should not be altered by MoBi (or PK-Sim) without user confirmation. Just opening them for viewing should not leave traces.

**Comment by msevestre** [comment]
The file is a database. The structure of the database has changed between two versions. There is no way around converting the database (which needs to happen physically) before loading the content of the project..so generally I agree, we try not to alter the file unless the user saves it. In those very rare cases however, there is no way around. The database structure has not changed in MoBi in over 4 years... So this issue is very rare. It could however happen if we change the structure again in the future

The only alternative I see would be to warn the user before loading the file (by reading some meta data somehow telling us that a conversion will happen). The warning would say that an irreversible conversion will happen with a prompt to continue, cancel or maybe even create a backup.

**Comment by UlrichSi** [comment]
That doesn't sound too good. :-(

A warning would be better than nothing, though. Another way could be that MoBi leaves a backup version of the original file (like Matlab's .asv files).

**Comment by tobiasK2001** [comment]
hm..., a warning with the option "continue and create untouched back up" would be my preferred option.

**Comment by StephanSchaller** [comment]
I agree. The backup is a must. While the user should have made backups already anyway, for Quality Assurance reasons, the file database version shouldn't just be updated irreversibly without the users consent.

A warning with a Matlab-style backup that will be restored if the user doesn't save the project sounds good.
